src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+    else {
+      $params['ProjectMetadata'][] = [
+        'MetadataKey' => 'Workflow Options',
+        'MetadataValue' => 'Localize Only',
+      ];
+    }
+
+    return $this->request('project', 'POST', $params);
+  }
+
+  /**
+   * Send the files to Thebigword.
+   *
+   * @param \Drupal\tmgmt\JobItemInterface $job_item
+   *   The Job.
+   */
+  private function sendFiles(JobItemInterface $job_item) {
+    $cache = \Drupal::cache('data');
+    /** @var \Drupal\tmgmt_file\Format\FormatInterface $xliff_converter */
+    $xliff_converter = \Drupal::service('plugin.manager.tmgmt_file.format')->createInstance('xlf');
+    $resource_uuids = array();
+
+    $job_item_id = $job_item->id();
+    $target_language = $job_item->getJob()->getRemoteTargetLanguage();
+    $cid = "tmgmt_thebigword:resource_id:$job_item_id:$target_language";
+    if ($cached = $cache->get($cid)) {
+      $file_id = $cached->data;
+    }
+    else {
+      $conditions = array('tjiid' => array('value' => $job_item_id));
+      $xliff = $xliff_converter->export($job_item->getJob(), $conditions);
+      $name = "JobID_{$job_item->getJob()->id()}_JobItemID_{$job_item_id}_{$job_item->getJob()->getSourceLangcode()}_{$target_language}";
+
+      $file_id = $this->uploadFileResource($xliff, $job_item->getJob(), $job_item_id, $name);
+      $this->sendPreviewUrl($job_item, $file_id, FALSE);
+      $cache->set($cid, $file_id, Cache::PERMANENT, $job_item->getCacheTags());
+    }
+    $resource_uuids[$job_item_id] = $file_id;
+  }
+
+  /**
+   * Creates a file resource at Thebigword.
+   *
+   * @param string $xliff
+   *   .XLIFF string to be translated. It is send as a file.
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   The Job.
+   * @param int $tjiid
+   *   The JobItem id.
+   * @param string $name
+   *   File name of the .XLIFF file.
+   *
+   * @return string Thebigword uuid of the resource.
+   *   Thebigword uuid of the resource.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  public function uploadFileResource($xliff, JobInterface $job, $tjiid, $name) {
+    $remote_mappings = RemoteMapping::loadByLocalData($job->id(), $tjiid, 'tmgmt_thebigword');
+    $remote_mapping = reset($remote_mappings);
+    $project_id = $remote_mapping->getRemoteIdentifier2();
+
+    /** @var \DateTime $required_by */
+    $required_by = $job->getSetting('required_by');
+    $required_by->setTimezone(new \DateTimeZone('UTC'));
+    $datetime = $required_by->format('Y-m-d\TH:i:s');
+    $form_params = [
+      'ProjectId' => $project_id,
+      'RequiredByDateUtc' => $datetime,
+      'SourceLanguage' => $job->getRemoteSourceLanguage(),
+      'TargetLanguage' => $job->getRemoteTargetLanguage(),
+      'FilePathAndName' => "$name.xliff",
+      'FileState' => 'TranslatableSource',
+      'FileData' => base64_encode($xliff),
+    ];
+    /** @var int $file_id */
+    $file_id = $this->request('file', 'POST', $form_params);
+
+    $files = $remote_mapping->getRemoteData('files');
```

Comment:
ah, so you use the remote mapping for this as well. Since you also return the ID, I think it would be better to do this in the parent.